spynnaker7/pyNN/models/pynn_projection.py

```diff
@@ -27,7 +32,11 @@ def __init__(
             synapse_dynamics=None, rng=None):
         # pylint: disable=too-many-arguments, protected-access
         synapse_dynamics_stdp = None
-        if synapse_dynamics is None:
+        if (synapse_dynamics is None or
+                (isinstance(synapse_dynamics,
+                 AbstractSynapseDynamicsStructural) and
+                 isinstance(synapse_dynamics.super,
+                 AbstractStaticSynapseDynamics))):
```

Comment:
I have this feeling that there ought to be a better way of writing these few lines… but I can't think of it. (Properly, we ought to enforce that there's a property from `AbstractSynapseDynamicsStructural`.)